docs/getting-started-objc-macos.md

```diff
@@ -0,0 +1,151 @@
+				var str = wc.DownloadString (url);
+				var json = JsonValue.Parse (str)["query"]["results"]["channel"]["item"]["condition"];
+				var result = new XAMWeatherResult (json["temp"], json["text"]);
+				return result;
+			}
+		}
+		catch (Exception ex) {
+			// Log some of the exception messages
+			Console.WriteLine (ex.Message);
+			Console.WriteLine (ex.InnerException?.Message);
+			Console.WriteLine (ex.InnerException?.InnerException?.Message);
+
+			return null;
+		}
+
+	}
+}
+
+public class XAMWeatherResult {
+	public string Temp { get; private set; }
+	public string Text { get; private set; }
+
+	public XAMWeatherResult (string temp, string text)
+	{
+		Temp = temp;
+		Text = text;
+	}
+}
+```
+
+You will notice that `Using System.Json;` is giving an error, in order to fix this we need to do the following:
+
+1. Double click on **References** folder.
+
+2. Click on **Packages** tab.
+
+3. Check **System.Json**.
+
+4. Click **Ok** button.
+
+Once the above is done all we need to do is build our .NET Assembly by clicking on *Build menu > Build All*  or ⌘+ b. Build Succesful message should appear in the top status bar.
+
+Now right click on *Weather* project node and select *Reveal in Finder,* now in finder go to *bin/Debug* folder, inside it you will find **Weather.dll.**
+
+## Using Embeddinator-4000
+
+If you installed Embeddinator-4000 using our pkg installer you should find **objcgen** in the following path `/Library/Frameworks/Xamarin.Embeddinator-4000/Commands/objcgen` , **objcgen** is the tool that we need to generate a native library from a .NET Assembly.
+
+Open terminal and `cd` into the folder containing Weather.dll and execute the following command:
+
+```shell
+/Library/Frameworks/Xamarin.Embeddinator-4000/Commands/objcgen --debug --outdir=output -c Weather.dll
+```
+
+Everything you need will be placed into **output** directory next to *Weather.dll*. If you have your own .NET Assembly just replace *Weather.dll* with it and follow the same steps above.
+
+## Using the generated output in an Xcode project
+
+Open Xcode and create a new **macOS Cocoa Application** and name it **MyWeather**. Right click on *MyWeather project node* and select *Add Files to "MyWeather"* and navigate to the **output** directory created by *Embeddinator-4000* and add the following files:
+
+* bindings.h
+* embeddinator.h
+* glib.h
+* mono-support.h
+* mono_embeddinator.h
+* objc-support.h
+* libWeather.dylib
+* Weather.dll 
+
+Make sure **Copy items if needed** is checked in the options panel of the file dialog.
+
+Now we need to make sure that **libWeather.dylib** and **Weather.dll** get into the app bundle. Click on *MyWeather project node* and select *Build Phases* tab and add a new *Copy Files Phase* next on *Destination* select **Frameworks** and add **libWeather.dylib**. Now add a new *Copy Files Phase* and on *Destination* select **Executables** now add **Weather.dll** and make sure *Code Sign on Copy* is checked.
```

Comment:
Now we need to make sure that **libWeather.dylib** and **Weather.dll** get into the app bundle. 
* Click on *MyWeather project node* 
* Select *Build Phases* tab
* Add a new *Copy Files Phase* 
* On *Destination* select **Frameworks** and add **libWeather.dylib**. 
* Add a new *Copy Files Phase* and on *Destination* select **Executables** 
* Now add **Weather.dll** and make sure *Code Sign on Copy* is checked.